src/App.tsx

```diff
@@ -1,12 +1,62 @@
+/* eslint-disable react/no-unused-state */
+/* eslint-disable no-console */
 import React from 'react';
 import './App.scss';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './type/Good';
+import { GoodsList } from './component/GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+type State = {
+  goods: Good[];
+};
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+  };
+
+  handler = (promise: Promise<Good[]>) => {
+    promise.then((goods: Good[]) => {
```

Comment:
unless a library or legacy codebase forces you to use then/catch, the better choice for readability and maintainability is async/await